src/vivarium/framework/state_machine.py

```diff
@@ -443,7 +448,9 @@ def extend(self, transitions: Iterable[Transition]) -> None:
     # Helper methods #
     ##################
 
-    def _normalize_probabilities(self, outputs, probabilities):
+    def _normalize_probabilities(
+        self, outputs: list[State | str], probabilities: NumericArray
```

Comment:
you could avoid some of this `| str` business by making "null_transition" a `State` instead of just a string, but i wasn't sure whether it was worth the hassle.